torch/csrc/distributed/c10d/init.cpp

```diff
@@ -3256,7 +3259,16 @@ for details.
           // shouldn't leak because of allocation in strdup.
           [](ncclConfig_t& self, const char* tmp) {
             self.netName = strdup(tmp);
-          });
+          })
+      .def(
+          "__copy__",
+          [](const ncclConfig_t& self) { return ncclConfig_t(self); })
+      .def(
+          "__deepcopy__",
+          [](const ncclConfig_t& self, const py::dict& memo) {
+            return ncclConfig_t(self);
+          },
+          py::arg("memo"));
```

Comment:
Wondering how we could make this look better? Can pybind find the default copy constructor of a struct to fulfill `__copy__`?

Reply:
I couldn't find a smarter way of doing this, and even within this file there are other instances which are done the same way:
https://github.com/pytorch/pytorch/blob/f45f483884ef524f2d8260dcf2b2ba2c203eab2f/torch/csrc/distributed/c10d/init.cpp#L869-L876